packages/datastore/src/entity.js

```diff
@@ -378,14 +384,70 @@ entity.entityFromEntityProto = entityFromEntityProto;
  * // }
  */
 function entityToEntityProto(entityObject) {
-  return {
+  var properties = entityObject.data;
+  var excludeFromIndexes = entityObject.excludeFromIndexes;
+
+  var entityProto = {
     key: null,
 
-    properties: Object.keys(entityObject).reduce(function(properties, key) {
-      properties[key] = entity.encodeValue(entityObject[key]);
-      return properties;
+    properties: Object.keys(properties).reduce(function(encoded, key) {
+      encoded[key] = entity.encodeValue(properties[key]);
+      return encoded;
     }, {})
   };
+
+  if (excludeFromIndexes && excludeFromIndexes.length > 0) {
+    excludeFromIndexes.forEach(function(excludePath) {
+      excludePathFromEntity(entityProto, excludePath);
+    });
+  }
+
+  return entityProto;
+
+  function excludePathFromEntity(entity, path) {
+    var arrayIndex = path.indexOf('[]');
```

Comment:
I feel like the bulk of this function could be modularized, I immediately thought of `dot-prop` but upon closer inspection that does not support Arrays. Is there a similar module that does? Or do you think we should consider making this a module?